src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyConverter.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace NuGet.Protocol
+{
+    public class PackageDependencyConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType) => objectType == typeof(Packaging.Core.PackageDependency);
+
+        public override bool CanWrite => false;
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            string id = null;
+            string version = null;
+
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
```

Comment:
I couldn't find a converter for it, but it seems simple enough to make. The range property name doesn't match the one on the PackageDependency object, but you can just nsj to handle that right?

Reply:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Converters/VersionRangeConverter.cs

This PR is still using Newtonsoft.json, not System.Text.Json.

Reply:
I tried using Newtonsoft.Json to automatically figure out `PackageDependency` by adding `JsonPropertyNames` to the class, I couldn't figure out why it wasn't working, it was complaining about the constructors not being correct (I did add the [JsonConstructor] property).

So, at the end I think I'm keeping this constructor and maybe it could be helpful in the future, it also helped me do unit tests. Also, I found that `JsonExtensions.JsonObjectSerializer` contains some of our converters, including the `VersionRangeConverter` and then just do `serializer.Deserialize<Packaging.Core.PackageDependency>(reader);`